### PR TITLE
Add GitHub Actions which publish to PyPI

### PR DESCRIPTION
**Pull request by mithro**
This pull request will add a GitHub Action which builds the fasm packages, including binary wheels for Linux, Mac and Windows and then upload them to PyPI.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,13 @@ project(parse_fasm)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/antlr4/runtime/Cpp/cmake)
```
Comment (mithro) [inline src/CMakeLists.txt:14]: @litghost -- I need some help here dealing with these compiler flags.

I would like to;
 - Add `-Wimplicit-fallthrough=0` if it is supported on non-Windows
 - Add `-Wno-attributes` if it is supported on non-Windows
 - Add `/std:c++latest` on Windows..

(IE I would like to get rid of the `add_flags` function in the setup.py Python file above and just have cmake correctly understand the flags.)

File: Makefile [hunk 2]
```diff
@@ -1,65 +1,66 @@
-# Copyright (C) 2017-2020  The SymbiFlow Authors.
```
Comment (mithro) [inline Makefile:1]: Reminder to myself -- Looks like I need to restore the make targets I seem to have accidentally removed.

File: src/CMakeLists.txt (hunk 1 above)
Comment (litghost) [reply src/CMakeLists.txt:14]: https://docs.microsoft.com/en-us/cpp/build/reference/compiler-options?view=msvc-160

Comment (litghost) [reply src/CMakeLists.txt:14]: https://en.cppreference.com/w/cpp/language/attributes/fallthrough

If ANTLR could emit `[[fallthrough]]`, `-Wimplicit-fallthrough=0` is not required.

Comment (litghost) [reply src/CMakeLists.txt:14]: PIC support might be possible via: https://cmake.org/cmake/help/v3.0/prop_tgt/POSITION_INDEPENDENT_CODE.html

Comment (litghost) [reply src/CMakeLists.txt:14]: CMake has explicit C++ standard select support: https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+name: make-env
+dependencies:
+  - python
+  - pip
+  - cmake
```
Comment (litghost) [inline environment.yml:5]: Needs SWIG?

File: Makefile (hunk 2 above)
Comment (mithro) [reply Makefile:1]: Fixed.

File: environment.yml (hunk 3 above)
Comment (mithro) [reply environment.yml:3]: Done.

File: src/CMakeLists.txt (hunk 1 above)
Comment (mithro) [reply src/CMakeLists.txt:14]: I'm going to defer this to #51

**Comment by mithro** [comment]
@litghost -- As this pull request seems far enough along that someone else can fix the remaining issues and it seems unlikely I'll have time to do it myself, I think we should merge and move forward. Thus, I have logged github issues about the remaining items.

**Pull request merged**